R/contribution_matrix.R

```diff
@@ -0,0 +1,588 @@
+        #...put 1 in the treatment column...
+        X[which(GetStudies(rownames(X)) == data_no_control$Study[i] & GetTreatments(rownames(X)) == data_no_control$Treatment[i]), which(GetTreatments(colnames(X)) == data_no_control$Treatment[i])] <- 1
+        #...put -1 in the control treatment column...
+        X[which(GetStudies(rownames(X)) == data_no_control$Study[i] & GetTreatments(rownames(X)) == data_no_control$Treatment[i]), which(GetTreatments(colnames(X)) == data_no_control$Control[i])] <- -1
+        #...put the covariate value in the treatment's covariate column...
+        X[which(GetStudies(rownames(X)) == data_no_control$Study[i] & GetTreatments(rownames(X)) == data_no_control$Treatment[i]), which(GetTreatments(colnames(X)) == paste0(data_no_control$Treatment[i], "_beta"))] <- covar_centered[data_no_control$Study[i]]
+        #...and put minus the covariate value in the control treatment's covariate column
+        X[which(GetStudies(rownames(X)) == data_no_control$Study[i] & GetTreatments(rownames(X)) == data_no_control$Treatment[i]), which(GetTreatments(colnames(X)) == paste0(data_no_control$Control[i], "_beta"))] <- -covar_centered[data_no_control$Study[i]]
+      }
+    }
+  }
+  
+  #Populate the design matrix for shared covariate parameters
+  if(cov_parameters == "Shared"){
+    for(i in 1:length(data_no_control$Study)){
+      #If the study contains the reference treatment...
+      if(data_no_control$Control[i] == reference){
+        #...put 1 in the treatment column...
+        X[which(GetStudies(rownames(X)) == data_no_control$Study[i] & GetTreatments(rownames(X)) == data_no_control$Treatment[i]), which(GetTreatments(colnames(X)) == data_no_control$Treatment[i])] <- 1
+        #...and put the covariate value in the covariate column
+        X[which(GetStudies(rownames(X)) == data_no_control$Study[i] & GetTreatments(rownames(X)) == data_no_control$Treatment[i]), which(GetTreatments(colnames(X)) == "B")] <- covar_centered[data_no_control$Study[i]]
+        
+        #If the study does not contain the reference treatment...
+      } else if(data_no_control$Control[i] != reference){
+        #...put 1 in the treatment column...
+        X[which(GetStudies(rownames(X)) == data_no_control$Study[i] & GetTreatments(rownames(X)) == data_no_control$Treatment[i]), which(GetTreatments(colnames(X)) == data_no_control$Treatment[i])] <- 1
+        #...put -1 in the control treatment column...
+        X[which(GetStudies(rownames(X)) == data_no_control$Study[i] & GetTreatments(rownames(X)) == data_no_control$Treatment[i]), which(GetTreatments(colnames(X)) == data_no_control$Control[i])] <- -1
+        #...and put 0 in the covariate column
+        X[which(GetStudies(rownames(X)) == data_no_control$Study[i] & GetTreatments(rownames(X)) == data_no_control$Treatment[i]), which(GetTreatments(colnames(X)) == "B")] <- 0
+      }
+    }
+  }
+  
+  return(X)
+}
+
+
+
+#' Create the Z matrix.
```

Comment:
What's a "Z matrix"? Could the roxygen description be expanded please?

Reply:
See comment about the X matrix.